In [114]:
# 기본 패키지 불러오기 

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns 
plt.style.use("seaborn")
sns.set(font_scale=1)
sns.set_style("whitegrid")

# 그래프를 노트북 안에 그리기 위해 설정
%matplotlib inline

# 맥 OS 폰트 깨짐 방지 코드
from matplotlib import rc
import matplotlib as mpl 
import matplotlib.font_manager as fm 

rc('font', family='AppleGothic')
plt.rcParams['axes.unicode_minus'] = False

## Inside Airbnb: London Data

### 데이터 불러오기

In [115]:
london_lists_raw = pd.read_csv('../../../../data_weekly/inside_airbnb/London/listings.csv')
london_lists_summary_raw = pd.read_csv('../../../../data_weekly/inside_airbnb/London/listings_outline.csv')
london_neighbourhoods_raw = pd.read_csv('../../../../data_weekly/inside_airbnb/London/neighbourhoods.csv')
london_reviews_raw = pd.read_csv('../../../../data_weekly/inside_airbnb/London/reviews.csv')
london_reviews_summary_raw = pd.read_csv('../../../../data_weekly/inside_airbnb/London/reviews_outline.csv')

In [116]:
london_lists = london_lists_raw.copy()
london_lists

id                                       listing_url  \
0                   312761               https://www.airbnb.com/rooms/312761   
1                    13913                https://www.airbnb.com/rooms/13913   
2                    15400                https://www.airbnb.com/rooms/15400   
3                   159736               https://www.airbnb.com/rooms/159736   
4                   165336               https://www.airbnb.com/rooms/165336   
...                    ...                                               ...   
90847  1115272743905612628  https://www.airbnb.com/rooms/1115272743905612628   
90848  1115272767893608505  https://www.airbnb.com/rooms/1115272767893608505   
90849  1115285644506865913  https://www.airbnb.com/rooms/1115285644506865913   
90850  1115338006572763275  https://www.airbnb.com/rooms/1115338006572763275   
90851  1115347885713265376  https://www.airbnb.com/rooms/1115347885713265376   

            scrape_id last_scraped       source  \
0      20240319050633   2024-03-21  city scrape   
1      20240319050633   2024-03-20  city scrape   
2      20240319050633   2024-03-20  city scrape   
3      20240319050633   2024-03-20  city scrape   
4      20240319050633   2024-03-21  city scrape   
...               ...          ...          ...   
90847  20240319050633   2024-03-19  city scrape   
90848  20240319050633   2024-03-19  city scrape   
90849  20240319050633   2024-03-21  city scrape   
90850  20240319050633   2024-03-20  city scrape   
90851  20240319050633   2024-03-19  city scrape   

                                                    name  \
0      Spacious Central London Apartment by Hoxton Sq...   
1                    Holiday London DB Room Let-on going   
2                    Bright Chelsea  Apartment. Chelsea!   
3       A double Room 5mins from King's College Hospital   
4                          Charming Flat in Notting Hill   
...                                                  ...   
90847                           Primrose Hill Studio #25   
90848                           Primrose Hill Studio #27   
90849                Quad Bedroom with External Bathroom   
90850                               Grosvenor Collection   
90851                            Comfy Home - Wood Green   

                                             description  \
0      Very central location, in the middle of Shored...   
1      My bright double bedroom with a large window h...   
2      Lots of windows and light.  St Luke's Gardens ...   
3      Calm sunny double room with a queen size bed a...   
4      A stylish apartment close to Portobello market...   
...                                                  ...   
90847  A place to spread out in with extra wiggle roo...   
90848  Two-storey living, unwind downstairs and rest ...   
90849  The Chapter Hotels group had been in the hotel...   
90850  This bright and spacious three-bedroom penthou...   
90851  Great for families and groups travelling toget...   

                                   neighborhood_overview  \
0      Everything is so convenient and the area is al...   
1      Finsbury Park is a friendly melting pot commun...   
2                                         It is Chelsea.   
3      We love that in Loughborough Junction we live ...   
4      Notting Hill has many cafes, bars and restaura...   
...                                                  ...   
90847                                                NaN   
90848                                                NaN   
90849                                                NaN   
90850  The area itself is a vibrant neighbourhood, of...   
90851                                                NaN   

                                             picture_url    host_id  ...  \
0      https://a0.muscache.com/pictures/miso/Hosting-...    1608226  ...   
1      https://a0.muscache.com/pictures/miso/Hosting-...      54730  ...   
2      https://a0.muscache.com/pictures/428392/462d26...   

In [117]:
london_lists.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90852 entries, 0 to 90851
Data columns (total 75 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   id                                            90852 non-null  int64  
 1   listing_url                                   90852 non-null  object 
 2   scrape_id                                     90852 non-null  int64  
 3   last_scraped                                  90852 non-null  object 
 4   source                                        90852 non-null  object 
 5   name                                          90852 non-null  object 
 6   description                                   87851 non-null  object 
 7   neighborhood_overview                         47521 non-null  object 
 8   picture_url                                   90842 non-null  object 
 9   host_id                                       90852 non-null 

In [118]:
london_lists.columns.to_frame().reset_index(drop=True).rename(columns={0:'column name'}).to_csv('./NameofColumns_rawdata.csv')

### 분석에서 사용하지 않을 컬럼 삭제

In [119]:
london_lists.columns

Index(['id', 'listing_url', 'scrape_id', 'last_scraped', 'source', 'name',
       'description', 'neighborhood_overview', 'picture_url', 'host_id',
       'host_url', 'host_name', 'host_since', 'host_location', 'host_about',
       'host_response_time', 'host_response_rate', 'host_acceptance_rate',
       'host_is_superhost', 'host_thumbnail_url', 'host_picture_url',
       'host_neighbourhood', 'host_listings_count',
       'host_total_listings_count', 'host_verifications',
       'host_has_profile_pic', 'host_identity_verified', 'neighbourhood',
       'neighbourhood_cleansed', 'neighbourhood_group_cleansed', 'latitude',
       'longitude', 'property_type', 'room_type', 'accommodates', 'bathrooms',
       'bathrooms_text', 'bedrooms', 'beds', 'amenities', 'price',
       'minimum_nights', 'maximum_nights', 'minimum_minimum_nights',
       'maximum_minimum_nights', 'minimum_maximum_nights',
       'maximum_maximum_nights', 'minimum_nights_avg_ntm',
       'maximum_nights_avg_ntm', 'ca

In [120]:
# 1차 선별한 컬럼 리스트 
# 1차 선별한 컬럼 27개 + 검증만 하고 삭제하기로 합의한 컬럼 3개('host_has_profile_pic','host_identity_verified', 'has_availability')
# 총 30개 컬럼 선별 

columns_selected = ['id', 'host_id', 'host_response_time', 'host_response_rate', 'host_acceptance_rate', 'host_is_superhost', 'host_total_listings_count', 'host_has_profile_pic', 'host_identity_verified', 'neighbourhood_cleansed', \
    'room_type', 'accommodates', 'bathrooms', 'bedrooms', 'beds', 'amenities', 'price', 'minimum_nights', 'maximum_nights', 'has_availability', 'number_of_reviews', 'number_of_reviews_l30d', 'review_scores_rating', 'review_scores_accuracy', \
    'review_scores_cleanliness', 'review_scores_checkin', 'review_scores_communication', 'review_scores_location', 'review_scores_value', 'reviews_per_month']
print(columns_selected)

['id', 'host_id', 'host_response_time', 'host_response_rate', 'host_acceptance_rate', 'host_is_superhost', 'host_total_listings_count', 'host_has_profile_pic', 'host_identity_verified', 'neighbourhood_cleansed', 'room_type', 'accommodates', 'bathrooms', 'bedrooms', 'beds', 'amenities', 'price', 'minimum_nights', 'maximum_nights', 'has_availability', 'number_of_reviews', 'number_of_reviews_l30d', 'review_scores_rating', 'review_scores_accuracy', 'review_scores_cleanliness', 'review_scores_checkin', 'review_scores_communication', 'review_scores_location', 'review_scores_value', 'reviews_per_month']


In [121]:
# 선별한 컬럼만 적용

london_lists = london_lists[columns_selected]
london_lists.head(10)

id  host_id  host_response_time host_response_rate  \
0  312761  1608226  within a few hours               100%   
1   13913    54730  within a few hours                86%   
2   15400    60302        within a day               100%   
3  159736   766056      within an hour               100%   
4  165336   761400      within an hour               100%   
5  313710  1612623                 NaN                NaN   
6  168359   648725                 NaN                NaN   
7   17402    67564      within an hour               100%   
8   24328    41759                 NaN                NaN   
9  170524   194769        within a day                67%   

  host_acceptance_rate host_is_superhost  host_total_listings_count  \
0                  98%                 t                        3.0   
1                  84%                 t                        5.0   
2                  40%                 f                        1.0   
3                  78%                 t                        5.0   
4                  92%                 t                        1.0   
5                  NaN                 f                        1.0   
6                  NaN                 f                        2.0   
7                  96%                 t                       32.0   
8                   0%                 f                        3.0   
9                  50%                 f                        9.0   

  host_has_profile_pic host_identity_verified  neighbourhood_cleansed  ...  \
0                    t                      t                 Hackney  ...   
1                    t                      t               Islington  ...   
2                    t                      t  Kensington and Chelsea  ...   
3                    t                      t                 Lambeth  ...   
4                    t                      t  Kensington and Chelsea  ...   
5                    t                      t                   Brent  ...   
6                    t                      f               Southwark  ...   
7                    t                      t             Westminster  ...   
8                    t                      t              Wandsworth  ...   
9                    t                      t             Westminster  ...   

  number_of_reviews  number_of_reviews_l30d  review_scores_rating  \
0               122                       0                  4.96   
1                41                       0                  4.80   
2                94                       0                  4.80   
3                96                       0                  4.72   
4               237                       2                  4.74   
5                10                       0                  4.88   
6                 9                       0                  4.63   
7                56                       0                  4.77   
8                94                       0                  4.90   
9                 0                       0                   NaN   

   review_scores_accuracy  review_scores_cleanliness review_scores_checkin  \
0                    4.97                       4.93                  4.96   
1                    4.73                       4.73                  4.75   
2                    4.85                       4.88                  4.88   
3                    4.79                       4.66                  4.75   
4                    4.88                       4.79                  4.94   
5                    4.67                       4.89                  4.89   
6                    4.75                       4.63                  4.88   
7                    4.83                       4.72                  4.72   
8                    4.89                       4.91                  4.90   
9                     NaN                        NaN                   NaN   

  review_scores_communication  review_scores_location  review_scores_value 

In [122]:
london_lists.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90852 entries, 0 to 90851
Data columns (total 30 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   id                           90852 non-null  int64  
 1   host_id                      90852 non-null  int64  
 2   host_response_time           57797 non-null  object 
 3   host_response_rate           57797 non-null  object 
 4   host_acceptance_rate         65213 non-null  object 
 5   host_is_superhost            90463 non-null  object 
 6   host_total_listings_count    90847 non-null  float64
 7   host_has_profile_pic         90847 non-null  object 
 8   host_identity_verified       90847 non-null  object 
 9   neighbourhood_cleansed       90852 non-null  object 
 10  room_type                    90852 non-null  object 
 11  accommodates                 90852 non-null  int64  
 12  bathrooms                    57894 non-null  float64
 13  bedrooms        

In [123]:
# 컬럼명 통일 

london_lists = london_lists.rename(columns={
    'id': '숙소_id',
    'host_id': '호스트_id',
    'host_response_time': '답변_평균시간',
    'host_response_rate': '문의_응답률',
    'host_acceptance_rate': '예약_수락률',
    'host_is_superhost': '슈퍼호스트',
    'host_total_listings_count': '숙소_수',
    'host_has_profile_pic' : '프로필_사진',
    'host_identity_verified' : '호스트_신원',
    'neighbourhood_cleansed': '숙소_지역',
    'room_type': '숙소_유형',
    'accommodates': '수용_인원수',
    'bathrooms': '욕실수',
    'bedrooms': '침실수',
    'beds': '침대수',
    'amenities': '편의시설',
    'price': '숙소_가격',
    'minimum_nights': '최소_숙박일',
    'maximum_nights': '최대_숙박일',
    'has_availability' : '예약_가능여부',
    'number_of_reviews': '리뷰수',
    'number_of_reviews_l30d': '30일_리뷰수',
    'review_scores_rating': '리뷰점수',
    'review_scores_accuracy': '숙소_정확성_리뷰점수',
    'review_scores_cleanliness': '숙소_청결도_리뷰점수',
    'review_scores_checkin': '숙소_체크인_리뷰점수',
    'review_scores_communication': '숙소_소통_리뷰점수',
    'review_scores_location': '숙소_위치_리뷰점수',
    'review_scores_value': '숙소_가격_리뷰점수',
    'reviews_per_month': '평균_리뷰수'
})

In [124]:
london_lists.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90852 entries, 0 to 90851
Data columns (total 30 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   숙소_id        90852 non-null  int64  
 1   호스트_id       90852 non-null  int64  
 2   답변_평균시간      57797 non-null  object 
 3   문의_응답률       57797 non-null  object 
 4   예약_수락률       65213 non-null  object 
 5   슈퍼호스트        90463 non-null  object 
 6   숙소_수         90847 non-null  float64
 7   프로필_사진       90847 non-null  object 
 8   호스트_신원       90847 non-null  object 
 9   숙소_지역        90852 non-null  object 
 10  숙소_유형        90852 non-null  object 
 11  수용_인원수       90852 non-null  int64  
 12  욕실수          57894 non-null  float64
 13  침실수          78118 non-null  float64
 14  침대수          57837 non-null  float64
 15  편의시설         90852 non-null  object 
 16  숙소_가격        57885 non-null  object 
 17  최소_숙박일       90852 non-null  int64  
 18  최대_숙박일       90852 non-null  int64  
 19  예약_가

### 정형 데이터 내 좋은 숙소의 기준
* 슈퍼호스트는 좋은 숙소의 기준에 부합할까? 

슈퍼호스트 기준
1. 숙박 10건 이상 호스팅 또는 3건의 예약에 걸쳐 총 100박 이상 호스팅
2. 응답률 90% 이상 유지
3. 예약 취소율 1% 미만 유지. 단, 정상참작이 가능한 상황에 따른 예약 취소는 제외
4. 전체 평점 4.8점 이상 유지(후기 작성 기한인 14일이 지나거나, 그전에라도 게스트와 호스트 양측이 모두 후기를 제출하면 후기는 슈퍼호스트 실적에 계산됩니다).

Q. 정형 데이터를 통해 머신러닝을 돌리는게 프로젝트 평가 항목 중 하나
* 우리가 잡은 주제 내에서 무슨 예측을 어떤 형태로 해야할지도 고민해야함 

# 결측치

### 답변_평균시간

In [125]:
london_lists['답변_평균시간'].unique()

array(['within a few hours', 'within a day', 'within an hour', nan,
       'a few days or more'], dtype=object)

In [126]:
london_lists['답변_평균시간'].isnull().sum()

33055

In [127]:
london_lists['답변_평균시간'].value_counts()

within an hour        38118
within a few hours     9620
within a day           6757
a few days or more     3302
Name: 답변_평균시간, dtype: int64

In [128]:
# 답변 평균 시간 컬럼 제거 
# >> 좋은 숙소의 조건 가설로 잡은 슈퍼호스트 기준과 무관 
# >> 제외시키고 추후에 검증만 하기로 

london_lists.drop(columns='답변_평균시간', inplace=True)

In [129]:
london_lists.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90852 entries, 0 to 90851
Data columns (total 29 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   숙소_id        90852 non-null  int64  
 1   호스트_id       90852 non-null  int64  
 2   문의_응답률       57797 non-null  object 
 3   예약_수락률       65213 non-null  object 
 4   슈퍼호스트        90463 non-null  object 
 5   숙소_수         90847 non-null  float64
 6   프로필_사진       90847 non-null  object 
 7   호스트_신원       90847 non-null  object 
 8   숙소_지역        90852 non-null  object 
 9   숙소_유형        90852 non-null  object 
 10  수용_인원수       90852 non-null  int64  
 11  욕실수          57894 non-null  float64
 12  침실수          78118 non-null  float64
 13  침대수          57837 non-null  float64
 14  편의시설         90852 non-null  object 
 15  숙소_가격        57885 non-null  object 
 16  최소_숙박일       90852 non-null  int64  
 17  최대_숙박일       90852 non-null  int64  
 18  예약_가능여부      86773 non-null  object 
 19  리뷰수 

### 문의_응답률

In [130]:
# 문의_응답률 컬럼 

london_lists['문의_응답률'].isnull().sum()

33055

In [131]:
# 문의 응답률이 not null값인 숙소 >> 에어비앤비 사이트에서 샘플 확인

condition_response_notnull = london_lists['문의_응답률'].notnull()
london_lists_raw[condition_response_notnull][['listing_url', 'name', 'host_response_rate']][:30]

# >> 사이트 숙소 상세 설명 페이지 내 문의 응답률 위치 확인 

listing_url  \
0   https://www.airbnb.com/rooms/312761   
1    https://www.airbnb.com/rooms/13913   
2    https://www.airbnb.com/rooms/15400   
3   https://www.airbnb.com/rooms/159736   
4   https://www.airbnb.com/rooms/165336   
7    https://www.airbnb.com/rooms/17402   
9   https://www.airbnb.com/rooms/170524   
10  https://www.airbnb.com/rooms/314164   
14   https://www.airbnb.com/rooms/34935   
15   https://www.airbnb.com/rooms/36299   
16  https://www.airbnb.com/rooms/173082   
17  https://www.airbnb.com/rooms/174132   
19  https://www.airbnb.com/rooms/315238   
21  https://www.airbnb.com/rooms/316137   
22   https://www.airbnb.com/rooms/36660   
23  https://www.airbnb.com/rooms/316864   
24   https://www.airbnb.com/rooms/38610   
25   https://www.airbnb.com/rooms/38950   
26   https://www.airbnb.com/rooms/38995   
27  https://www.airbnb.com/rooms/178519   
29  https://www.airbnb.com/rooms/181664   
30  https://www.airbnb.com/rooms/318287   
31  https://www.airbnb.com/rooms/182802   
33   https://www.airbnb.com/rooms/41445   
34  https://www.airbnb.com/rooms/183128   
37  https://www.airbnb.com/rooms/320629   
38   https://www.airbnb.com/rooms/41712   
41  https://www.airbnb.com/rooms/198258   
42  https://www.airbnb.com/rooms/321411   
43  https://www.airbnb.com/rooms/198279   

                                                 name host_response_rate  
0   Spacious Central London Apartment by Hoxton Sq...               100%  
1                 Holiday London DB Room Let-on going                86%  
2                 Bright Chelsea  Apartment. Chelsea!               100%  
3    A double Room 5mins from King's College Hospital               100%  
4                       Charming Flat in Notting Hill               100%  
7                  Fab 3-Bed/2 Bath & Wifi: Trendy W1               100%  
9                            STUNNING APT IN ST JAMES                67%  
10                        Comfy Family Home in London                 0%  
14     Bright 1 Bedroom Apartment w 2 beds Brick Lane                96%  
15                Kew Gardens 3BR house in cul-de-sac               100%  
16                The Residential Suite Above Gallery               100%  
17                Large Bedroom with EnSuite Bathroom               100%  
19                      Stylish 2 bedroom London flat               100%  
21                       Double Room by Camden Market                97%  
22                    You are GUARANTEED to love this               100%  
23                            Trendy London Apartment               100%  
24                               CHARMING FAMILY HOME               100%  
25   Room 1 Large Double Bedroom - front ground floor                33%  
26           SPACIOUS ROOM IN CONTEMPORARY STYLE FLAT               100%  
27                  Newly refurbished 2-bed apartment               100%  
29  Stunning 2 bedroom Fulham Riverside Flat Free ...               100%  
30        safe and spacious room in comfy family home               100%  
31                 Quiet Double Room - Good for walks               100%  
33  2 Double bed apartment in quiet area North London               100%  
34            Private bathroom & spacious double room               100%  
37                    Hotel-style room in Camden Town               100%  
38  Room with a view, shared flat,  central  Bankside               100%  
41                    Penthouse Living in East London               100%  
42                     Fantastic location Double room               100%  
43                  London Whitechapel Room with WIFI               100%

응답률이 나와있는 숙소도 현재 사이트 내 문의 응답률과 다른 경우도 있음 
* https://www.airbnb.com/rooms/170524
* 66% > 33 %

In [132]:
# 문의 응답률이 null값인 숙소 >> 에어비앤비 사이트에서 샘플 확인

condition_response_null = london_lists['문의_응답률'].isnull()
london_lists_raw[condition_response_null][['listing_url', 'name']]

listing_url  \
5                   https://www.airbnb.com/rooms/313710   
6                   https://www.airbnb.com/rooms/168359   
8                    https://www.airbnb.com/rooms/24328   
11                  https://www.airbnb.com/rooms/170702   
12                   https://www.airbnb.com/rooms/33332   
...                                                 ...   
90844  https://www.airbnb.com/rooms/1115272705566417319   
90845  https://www.airbnb.com/rooms/1115272736222611647   
90846  https://www.airbnb.com/rooms/1115272737028206976   
90847  https://www.airbnb.com/rooms/1115272743905612628   
90848  https://www.airbnb.com/rooms/1115272767893608505   

                                                    name  
5                    Pretty Home Kensal Green NW10 1 bed  
6          Stunning 2 Bed Apartment in Elephant & Castle  
8      Battersea live/work artist house, garden communal  
11                       A Double Room with  River Views  
12        Beautiful Ensuite Richmond-upon-Thames borough  
...                                                  ...  
90844                           Primrose Hill Studio #29  
90845                           Primrose Hill Studio #28  
90846                           Primrose Hill Studio #26  
90847                           Primrose Hill Studio #25  
90848                           Primrose Hill Studio #27  

[33055 rows x 2 columns]

문의 응답률이 null값'인 숙소의 경우의 수 (가설)

1. 숙소 이용이 없어서 문의 또한 없기 때문에
2. 실제로 문의 응답을 안하는 경우 (카운팅을 하지 못하기 때문)

<br />

3. 에어비앤비 호스트 정보에 문의 응답률이 누락된 경우 
4. 데이터를 가져오면서 누락된 경우 및 기타 오류 

In [133]:
# 문의 응답률이 null 값인 숙소의 리뷰수와 리뷰 점수 조회
# 실제로 문의가

london_lists[condition_response_null][['리뷰수', '리뷰점수']]
london_lists[condition_response_null][['리뷰수', '리뷰점수']].sort_values(by='리뷰수', ascending=False)

리뷰수  리뷰점수
6874   711  4.87
885    575  4.95
1144   559  4.89
536    548  4.60
967    400  4.63
...    ...   ...
25321    0   NaN
25322    0   NaN
25334    0   NaN
25340    0   NaN
90848    0   NaN

[33055 rows x 2 columns]

리뷰가 많은데 문의 응답이 없는 경우가 있네?
* 이 케이스가 어떤 케이스인지 실제 사이트에서 확인해보자

In [134]:
london_lists_raw[condition_response_null][['listing_url', 'name']].shape

(33055, 2)

In [135]:
london_lists_raw[condition_response_null][['listing_url', 'name', 'number_of_reviews', 'review_scores_rating']].sort_values(by='number_of_reviews', ascending=False)[:30]

listing_url  \
6874    https://www.airbnb.com/rooms/9046849   
885      https://www.airbnb.com/rooms/877864   
1144    https://www.airbnb.com/rooms/1296836   
536      https://www.airbnb.com/rooms/502190   
967      https://www.airbnb.com/rooms/756481   
4030    https://www.airbnb.com/rooms/5878880   
20       https://www.airbnb.com/rooms/315658   
1087    https://www.airbnb.com/rooms/1117720   
8819   https://www.airbnb.com/rooms/11226035   
13151  https://www.airbnb.com/rooms/15855571   
8593   https://www.airbnb.com/rooms/10833014   
317      https://www.airbnb.com/rooms/397504   
804      https://www.airbnb.com/rooms/715564   
841      https://www.airbnb.com/rooms/726669   
9876   https://www.airbnb.com/rooms/12444228   
825      https://www.airbnb.com/rooms/720633   
985     https://www.airbnb.com/rooms/1044873   
840      https://www.airbnb.com/rooms/725370   
1322    https://www.airbnb.com/rooms/1626486   
819      https://www.airbnb.com/rooms/717220   
3560    https://www.airbnb.com/rooms/5041618   
21757  https://www.airbnb.com/rooms/23367928   
3276    https://www.airbnb.com/rooms/4772751   
12087  https://www.airbnb.com/rooms/14521008   
8993   https://www.airbnb.com/rooms/11419334   
9372   https://www.airbnb.com/rooms/11966878   
1093    https://www.airbnb.com/rooms/1185611   
28981  https://www.airbnb.com/rooms/32440623   
8546   https://www.airbnb.com/rooms/10706762   
21652  https://www.airbnb.com/rooms/23335713   

                                                    name  number_of_reviews  \
6874         Near HEATHROW AIRPORT, London, double room.                711   
885                   Bright guest room in gay household                575   
1144         Covent Garden Flat Listed Building, London.                559   
536     Large Double room private bath near Kilburn tube                548   
967                     Brand new studio in Victoria 15A                400   
4030     INDEPENDENT COSY LITTLE HOUSE IN CENTRAL LONDON                397   
20                    Double Room nr Oval Vauxhall Tubes                389   
1087                 Central London:  Sunny single room!                369   
8819    Tranquil comfortable single bedroom/females only                357   
13151  LHR,free travelzone,UB35BD vPrivate,vClean,Gar...                354   
8593                  Charming apartment, Central London                350   
317                                         Casa Chelsea                336   
804                      Brand new studio in Victoria 8A                332   
841                     Brand new studio in Victoria 11A                330   
9876    Room close to Central London - Wi-Fi and TV inc.                329   
825                   Double studio + patio, Victoria 2A                328   
985    Penthouse Room with Views over Park, near Station                308   
840                     Brand new studio in Victoria 12A                306   
1322   Private bathroom. Lovely area of North London N10                306   
819                         Double Studio in Victoria 5A                302   
3560   Pimlico Victoria Double Room Georgian Garden Flat                289   
21757  Charming Apartment in Central London - Farring...                289   
3276      Stylish 2 Bedroom Flat Central London *ZONE 1*                288   
12087       Beautiful, Victorian Room Facing The Garden.                286   
8993                              CAVENDISH  Apartment+1                285   
9372   Large single room with a big cosy bed with ens...                281   
1093                 Central London Zone1 Big Doubleroom                276   
28981                        Double room in Islington N1                275   
8546                  Clean and quite room with a garden                273   
21652      Master bedroom near Heathrow with an en-suite                270   

       review_scores_rating  
6874                   4.87  
885                    4.95  
1144

슈퍼호스트가 아닌 경우 문의 응답률이 호스트 정보에 나오질 않나? 
*  확인 필요

슈퍼호스트 컬럼 확인

In [136]:
# 슈퍼호스트 컬럼 null값 확인 
london_lists_raw['host_is_superhost'].isnull().sum()

389

In [137]:
# 슈퍼호스트 컬럼 유니크값 확인 
london_lists_raw['host_is_superhost'].unique()

array(['t', 'f', nan], dtype=object)

In [138]:
# 슈퍼호스트 컬럼 값별 개수 확인
# 슈퍼호스트 값이 t : 14,251
# 슈퍼호스트 값이 f : 76,212

condition_superhost_t = london_lists_raw['host_is_superhost'] == 't'
condition_superhost_f = london_lists_raw['host_is_superhost'] == 'f'
print('슈퍼호스트인 숙소 ')
print(london_lists_raw[condition_superhost_t].shape)
print()
print('슈퍼호스트가 아닌 숙소')
print(london_lists_raw[condition_superhost_f].shape)

슈퍼호스트인 숙소 
(14251, 75)

슈퍼호스트가 아닌 숙소
(76212, 75)


* 문의응답률 null 값 개수 : 33,055 개
* 슈퍼호스트가 아닌 숙소 : 76,212 개

<br />

*  슈퍼호스트가 아닌 경우에도 응답률 수집이 가능하긴 함!

리뷰가 많은데 문의응답률이 null값인 숙소의 슈퍼호스트 여부 확인

In [139]:
london_lists_raw[condition_response_null]['host_is_superhost'].unique()

array(['f', 't', nan], dtype=object)

In [140]:
london_lists_raw[condition_response_null][['listing_url', 'name', 'number_of_reviews', 'review_scores_rating', 'host_is_superhost']].sort_values(by='number_of_reviews', ascending=False)[:50]

listing_url  \
6874    https://www.airbnb.com/rooms/9046849   
885      https://www.airbnb.com/rooms/877864   
1144    https://www.airbnb.com/rooms/1296836   
536      https://www.airbnb.com/rooms/502190   
967      https://www.airbnb.com/rooms/756481   
4030    https://www.airbnb.com/rooms/5878880   
20       https://www.airbnb.com/rooms/315658   
1087    https://www.airbnb.com/rooms/1117720   
8819   https://www.airbnb.com/rooms/11226035   
13151  https://www.airbnb.com/rooms/15855571   
8593   https://www.airbnb.com/rooms/10833014   
317      https://www.airbnb.com/rooms/397504   
804      https://www.airbnb.com/rooms/715564   
841      https://www.airbnb.com/rooms/726669   
9876   https://www.airbnb.com/rooms/12444228   
825      https://www.airbnb.com/rooms/720633   
985     https://www.airbnb.com/rooms/1044873   
840      https://www.airbnb.com/rooms/725370   
1322    https://www.airbnb.com/rooms/1626486   
819      https://www.airbnb.com/rooms/717220   
3560    https://www.airbnb.com/rooms/5041618   
21757  https://www.airbnb.com/rooms/23367928   
3276    https://www.airbnb.com/rooms/4772751   
12087  https://www.airbnb.com/rooms/14521008   
8993   https://www.airbnb.com/rooms/11419334   
9372   https://www.airbnb.com/rooms/11966878   
1093    https://www.airbnb.com/rooms/1185611   
28981  https://www.airbnb.com/rooms/32440623   
8546   https://www.airbnb.com/rooms/10706762   
21652  https://www.airbnb.com/rooms/23335713   
199      https://www.airbnb.com/rooms/358360   
273      https://www.airbnb.com/rooms/381467   
8505   https://www.airbnb.com/rooms/10546640   
6364    https://www.airbnb.com/rooms/8127220   
1999    https://www.airbnb.com/rooms/2841245   
869      https://www.airbnb.com/rooms/739774   
5010    https://www.airbnb.com/rooms/6843786   
844      https://www.airbnb.com/rooms/726950   
13573  https://www.airbnb.com/rooms/16256750   
12824  https://www.airbnb.com/rooms/15333989   
777      https://www.airbnb.com/rooms/710203   
9999   https://www.airbnb.com/rooms/12613250   
813      https://www.airbnb.com/rooms/717171   
6062    https://www.airbnb.com/rooms/7876250   
16802  https://www.airbnb.com/rooms/19455388   
3629    https://www.airbnb.com/rooms/5081461   
871      https://www.airbnb.com/rooms/739825   
20389  https://www.airbnb.com/rooms/21988961   
12015  https://www.airbnb.com/rooms/14239782   
4651    https://www.airbnb.com/rooms/6532498   

                                                    name  number_of_reviews  \
6874         Near HEATHROW AIRPORT, London, double room.                711   
885                   Bright guest room in gay household                575   
1144         Covent Garden Flat Listed Building, London.                559   
536     Large Double room private bath near Kilburn tube                548   
967                     Brand new studio in Victoria 15A                400   
4030     INDEPENDENT COSY LITTLE HOUSE IN CENTRAL LONDON                397   
20                    Double Room nr Oval Vauxhall Tubes                389   
1087                 Central London:  Sunny single room!                369   
8819    Tranquil comfortable single bedroom/females only                357   
13151  LHR,free travelzone,UB35BD vPrivate,vClean,Gar...                354   
8593                  Charming apartment, Central London                350   
317                                         Casa Chelsea                336   
804                      Brand new studio in Victoria 8A                332   
841                     Brand new studio in Victoria 11A                330   
9876    Room close to Central London - Wi-Fi and TV inc.                329   
825                   Double studio + patio, Victoria 2A                328   
985    Penthouse Room with Views over Park, near Station                308   
840                     Brand new studio in Victoria 12A                306   
1322   Private bathroom. Lovely area of North London N10                306   
819 

* 응답률 null값 중 리뷰가 많은 숙소는 슈퍼호스트가 아니다.'는 가설은 기각. 
* 응답률 null값 중 리뷰가 많은 숙소의 대부분의 경우 슈퍼호스트가 아니다.' 이거는 맞는거 같은데? 

<br />

* 우선 응답률 null 값을 처리하는 방법은 투 트랙으로 가는 것이 좋을 듯.
    1. 리뷰 수가 많은 숙소의 경우 평균 또는 중위수로 처리
    2. 리뷰 수가 없는 숙소의 경우는 삭제하는 것이 맞지 않을까? 

근데 리뷰가 없는 애들은 분석할게 없으니까 먼저 정리하는게 맞지 않을까? 

우선은 중위수랑 평균 먼저 봐보자.

In [141]:
# 문의 응답률 >> 타입 변경 >> float

london_lists['문의_응답률'] = london_lists['문의_응답률'].str.rstrip('%').astype('float')
london_lists['문의_응답률'].head()

0    100.0
1     86.0
2    100.0
3    100.0
4    100.0
Name: 문의_응답률, dtype: float64

In [142]:
london_lists['문의_응답률'].describe()

count    57797.000000
mean        91.930360
std         21.931639
min          0.000000
25%         98.000000
50%        100.000000
75%        100.000000
max        100.000000
Name: 문의_응답률, dtype: float64

응답률 > 호스트 단위로 측정됨 (에어비앤비 CS 문의)

슈퍼호스트인데 문의 응답률이 없는 경우는 무엇인가? (에어비앤비 CS 팀에서 답변은 어렵다고 함)

슈퍼호스트  > 호스트 아이디로 묶어서 숙소 별로 일치하는지도 확인해야
